spec.md

```diff
@@ -582,6 +582,20 @@ message ControllerPublishVolumeRequest {
   // Whether to publish the volume in readonly mode. This field is
   // REQUIRED.
   bool readonly = 5;
+
+  // OPTIONAL field for storing principal information to be used in
+  // this publishing of this volume. If the principal is present the
+  // plugin MAY ensure all access to the volumes contents through
+  // this mount is mapped to the identity specified by the principal.
+  // If no principal is provided the access ids are not mapped but handed
+  // through untouched.
+  // As this field is specific to a given publishing of a volume
+  // containers in a container hierarchy have to either have their own
+  // individual volume publishings or a parent container MAY provide
+  // access to his volume mount throught some CO specific internal
+  // mechanism. In the latter case still all access through that volume
+  // mount is mapped to the given principal, as in a standalone container.
+  Principal accessed_as = 6;
```

Comment:
I understand your point, however the passed credential could be to the storage plugin itself  (such as kereberos ticket/ tokens/ etc) and thus needs to be send to the storage plugin. 
Many storage plugins may use the headless notion where there are practically no controller requests. You can have credential on both levels (controller and node) but this will only introduce the need to have multiple controllerPublish as well.
The same volume may be used by different containers each one has different credentials (working on behalf of different users) thus the storage plugin requires this information on a per publish (nodepublish)